File: decompiled/3.19.1/be/ehealth/businessconnector/therlink/domain/requests/PutTherapeuticLinkRequest.java
package be.ehealth.businessconnector.therlink.domain.requests;

import be.ehealth.businessconnector.therlink.domain.Author;
import be.ehealth.businessconnector.therlink.domain.Proof;
import be.ehealth.businessconnector.therlink.domain.TherapeuticLink;
import be.ehealth.businessconnector.therlink.domain.TherapeuticLinkRequestType;
import java.util.Date;
import org.joda.time.DateTime;

public class PutTherapeuticLinkRequest extends TherapeuticLinkRequestType {
   private static final long serialVersionUID = 1L;

   /** @deprecated */
   @Deprecated
   public PutTherapeuticLinkRequest(String id, Date date, Author author, TherapeuticLink link, Proof... proofs) {
      this(new DateTime(date), id, author, link, proofs);
   }

   public PutTherapeuticLinkRequest(DateTime date, String id, Author author, TherapeuticLink link, Proof... proofs) {
      super(id, date, author, link, proofs);
   }

   /** @deprecated */
   @Deprecated
   public PutTherapeuticLinkRequest(String id, Date date, Author author) {
      this(new DateTime(date), id, author);
   }

   public PutTherapeuticLinkRequest(DateTime date, String id, Author author) {
      super(id, (DateTime)date, author, (TherapeuticLink)null, (Proof[])null);
   }
}

File: src/main/kotlin/org/taktik/connector/business/therlink/domain/requests/PutTherapeuticLinkRequest.kt
package org.taktik.connector.business.therlink.domain.requests

import org.joda.time.DateTime
import org.taktik.connector.business.therlink.domain.Author
import org.taktik.connector.business.therlink.domain.Proof
import org.taktik.connector.business.therlink.domain.TherapeuticLink
import org.taktik.connector.business.therlink.domain.TherapeuticLinkRequestType
import java.util.*

class PutTherapeuticLinkRequest(id: String, dateTime: DateTime, author: Author, link: TherapeuticLink? = null, proofs: List<Proof?> = ArrayList()) : TherapeuticLinkRequestType(dateTime, id, author, link, proofs) {


	@Deprecated("")
	constructor(id: String, date: Date, author: Author, link: TherapeuticLink, vararg proofs: Proof?) : this(id, DateTime(date), author, link, proofs.toList())
	constructor(date: DateTime, id: String, author: Author, link: TherapeuticLink, vararg proofs: Proof?) : this(id, date, author, link, proofs.toList())

	@Deprecated("")
	constructor(id: String, date: Date, author: Author) : this(DateTime(date), id, author)
	constructor(date: DateTime, id: String, author: Author) : this(id, date, author)

	companion object {
		private val serialVersionUID = 1L
	}
}
